python/taichi/lang/ast/ast_transformer.py

```diff
@@ -476,17 +476,68 @@ def build_UnaryOp(ctx, node):
         node.ptr = op(node.operand.ptr)
         return node
 
+    @staticmethod
+    def build_short_circuit_and(operands):
+        if len(operands) == 1:
+            return operands[0].ptr
```

Comment:
Should we convert it to a bool value? Maybe by ti.logical_not(ti.logical_not(operands[0].ptr)) or something like that?

Reply:
Chi IR will already give a compile error at wrong types, so I think this is unnecessary?

Reply:
In rare circumstances there might be some people who writes codes like `a = 123 + (1 and 2)`, the value of `1 and 2` should be `1`(`True`) instead of `2`. IIUC in the current implementation the value of `1 and 2` is the value of the last operands which is `2`?

Reply:
Ah yes, this is indeed the case. It is also Python semantics (`1 and 2 == 2`), but normal (non short-circuiting) taichi semantics is casting that into boolean, so I'll change this.

Reply:
Your comment definitely makes sense. However, I have to say that the result of bool expressions in Taichi is undefined for now. For example, having the following code snippet,
```python
@ti.kernel
def func(a: ti.i32, b: ti.i32) -> ti.i32:
    return a and b
```
, you'll get `func(2, 3) == 2`. This is because Taichi uses `bitwise and` to implement `logical and`.

Therefore, I think this is not an issue for this PR. Instead, I think we need to open another issue to discuss what is the expected behavior of bool expressions in Taichi.

Reply:
We can discuss this in #3635.